samples/Common.c

```diff
@@ -864,9 +864,12 @@ STATUS getIceCandidatePairStatsCallback(UINT32 timerId, UINT64 currentTime, UINT
 
     pSampleConfiguration->rtcIceCandidatePairMetrics.requestedTypeOfStats = RTC_STATS_TYPE_CANDIDATE_PAIR;
 
-    // We need to execute this under the object lock due to race conditions that it could pose
-    MUTEX_LOCK(pSampleConfiguration->sampleConfigurationObjLock);
-    locked = TRUE;
+    // Use MUTEX_TRYLOCK to avoid possible dead lock when canceling timerQueue
+    if (!MUTEX_TRYLOCK(pSampleConfiguration->sampleConfigurationObjLock)) {
+        return retStatus;
+    } else {
```

Comment:
No need for else clause - just set the locked = TRUE as you are locked at this stage.

Reply:
thought so but preferred to copy & paste the code from pregenerateCertTimerCallback()
should i change both?

Reply:
Same comment. It's a minor issue that I am OK with. These elements were put in place and followed in the main library code for a reason of readability and better forward compatibility. I think it's up to you to decide.

Reply:
same, then lets keep the copy & pasted code